### ISSUE TITLE
Error encountered with n8n Form Trigger in queue mode utilizing separate webhook processors

### ISSUE THREAD

**Issue by dominikmukrecki** [n8n-io/n8n#7538]
**Description of the Issue:**

When utilizing the n8n Form Trigger in a queue mode with separate webhook processors, the system encounters errors while attempting to execute webhooks. Specifically, testing the webhook returns a 404 error, while executing the production webhook leads to a 500 error with the following error message: `{"code":0,"message":"Workflow Webhook Error: Workflow could not be started!"}`.

**Steps to Replicate:**

1. Create a new workflow within n8n.
2. Incorporate an n8n Form Trigger node into the workflow.
3. Activate the workflow.
4. Execute a GET request to the provided production link.

**Expected Outcome:**

The anticipation was for the webhook to process successfully without any errors.

**Environment Details:**

- **Operating System:** Docker container for n8n, tag "next"
- **n8n Version:** 1.14.2
- **Database System:** PostgreSQL
- **Operation Mode:** Queue

**Comment by dominikmukrecki** [comment]
[screencast-github.com-2023.10.27-13_53_30.webm](https://github.com/n8n-io/n8n/assets/64379788/4e3f998b-3d06-436c-858a-69ee20e0ca5d)

**Comment by Joffcom** [comment]
Hey @dominikmukrecki,

I have only managed to reproduce part of this, If I create a form my n8n worker doesn't appear to load the form. To get this fixed I have created an internal ticket which we will be tracking as `NODE-888`.

The other you are having with the preview I am not seeing, Does your browser load the popup window at all or is it being blocked?

**Comment by dominikmukrecki** [comment]
_Does your browser load the popup window at all or is it being blocked?_

@Joffcom yes, but I receive:

`Cannot GET /webhook-test/f9a4fd76-9cee-4169-9824-0d0a7b57448d/n8n-form`

**Comment by Joffcom** [comment]
Hey @dominikmukrecki,

Interesting, Which config options do you have set for your main instance of n8n?

**Comment by dominikmukrecki** [comment]
```
version: "3.9"

x-shared: &shared
  image: n8nio/n8n:${N8N_TAG}
  user: root:root
  depends_on:
    - redis
  volumes:
    - data:/home/node/.n8n
    - /mnt:/home/mnt
  networks:
    - default
  tmpfs:
    - /tmp
  environment:
    - DOMAIN_NAME=${DOMAIN_NAME}
    - SUBDOMAIN=${SUBDOMAIN}
    - N8N_HOST=${N8N_HOST}
    - WEBHOOK_URL=${WEBHOOK_URL}
    - WEBHOOK_TUNNEL_URL=${WEBHOOK_TUNNEL_URL}
    - SSL_EMAIL=${SSL_EMAIL}
    - DB_TYPE=${DB_TYPE}
    - DB_POSTGRESDB_DATABASE=${DB_POSTGRESDB_DATABASE}
    - DB_POSTGRESDB_HOST=${DB_POSTGRESDB_HOST}
    - DB_POSTGRESDB_USER=${DB_POSTGRESDB_USER}
    - DB_POSTGRESDB_PASSWORD=${DB_POSTGRESDB_PASSWORD}
    - POSTGRES_USER=${POSTGRES_USER}
    - POSTGRES_PASSWORD=${POSTGRES_PASSWORD}
    - POSTGRES_DB=${POSTGRES_DB}
    - POSTGRES_TAG=${POSTGRES_TAG}
    - EXECUTIONS_MODE=${EXECUTIONS_MODE}
    - QUEUE_BULL_REDIS_HOST=${QUEUE_BULL_REDIS_HOST}
    - QUEUE_HEALTH_CHECK_ACTIVE=${QUEUE_HEALTH_CHECK_ACTIVE}
    - REDIS_TAG=${REDIS_TAG}
    - N8N_BASIC_AUTH_ACTIVE=${N8N_BASIC_AUTH_ACTIVE}
    - N8N_BASIC_AUTH_USER=${N8N_BASIC_AUTH_USER}
    - N8N_BASIC_AUTH_PASSWORD=${N8N_BASIC_AUTH_PASSWORD}
    - N8N_BLOCK_ENV_ACCESS_IN_NODE=${N8N_BLOCK_ENV_ACCESS_IN_NODE}
    - N8N_DEFAULT_BINARY_DATA_MODE=${N8N_DEFAULT_BINARY_DATA_MODE}
    - N8N_PAYLOAD_SIZE_MAX=${N8N_PAYLOAD_SIZE_MAX}
    - N8N_PORT=${N8N_PORT}
    - N8N_PROTOCOL=${N8N_PROTOCOL}
    - N8N_TAG=${N8N_TAG}
    - N8N_DEFAULT_LOCALE=${N8N_DEFAULT_LOCALE}
    - NODE_ENV=${NODE_ENV}
    - NODE_FUNCTION_ALLOW_BUILTIN=${NODE_FUNCTION_ALLOW_BUILTIN}
    - NODE_FUNCTION_ALLOW_EXTERNAL=${NODE_FUNCTION_ALLOW_EXTERNAL}
    - GENERIC_TIMEZONE=${GENERIC_TIMEZONE}
    - EXECUTIONS_DATA_MAX_AGE=${EXECUTIONS_DATA_MAX_AGE}
    - EXECUTIONS_DATA_PRUNE=${EXECUTIONS_DATA_PRUNE}
    - EXECUTIONS_DATA_SAVE_MANUAL_EXECUTIONS=${EXECUTIONS_DATA_SAVE_MANUAL_EXECUTIONS}
    - EXECUTIONS_DATA_SAVE_ON_ERROR=${EXECUTIONS_DATA_SAVE_ON_ERROR}
    - EXECUTIONS_DATA_SAVE_ON_PROGRESS=${EXECUTIONS_DATA_SAVE_ON_PROGRESS}
    - EXECUTIONS_DATA_SAVE_ON_SUCCESS=${EXECUTIONS_DATA_SAVE_ON_SUCCESS}
    - EXECUTIONS_DATA_PRUNE_MAX_COUNT=${EXECUTIONS_DATA_PRUNE_MAX_COUNT}
    - N8N_EMAIL_MODE=${N8N_EMAIL_MODE}
    - CERT_RESOLVER=${CERT_RESOLVER}
    - N8N_ENCRYPTION_KEY=${N8N_ENCRYPTION_KEY}
x-database-shared: &database-shared
  restart: always

services:
  main:
    <<: *shared
    entrypoint: ["sh", "-c", "exec /bin/sh -c 'n8n start'"]
    depends_on:
      - redis
    deploy:
      mode: replicated
      placement:
        constraints:
          - node.role == worker
      resources:
        limits:
          memory: 256M
        reservations:
          memory: 128M
      update_config:
        parallelism: 1
        order: start-first
        delay: 30s
        
  worker:
    <<: *shared
    entrypoint: ["sh", "-c", "sleep 5 && exec /bin/sh -c 'n8n worker'"]
    depends_on:
      - main
      - redis
    deploy:
      mode: replicated
      replicas: 6
      placement:
        constraints:
          - node.role == worker
      resources:
        limits:
          memory: 256M
        reservations:
          memory: 128M
      update_config:
        parallelism: 1
        order: start-first
        delay: 30s
      
  webhook:
    <<: *shared
    entrypoint: ["sh", "-c", "sleep 5 && exec /bin/sh -c 'n8n webhook'"]
    expose:
      - 5678
    depends_on:
      - main
      - redis
    deploy:
      mode: replicated
      replicas: 6
      placement:
        constraints:
          - node.role == worker
      resources:
        limits:
          memory: 128M
        reservations:
          memory: 64M
      update_config:
        parallelism: 1
        order: start-first
        delay: 30s

  redis:
    <<: *database-shared
    image: redis:${REDIS_TAG}
    volumes:
      - redis_data:/data
    deploy:
      mode: replicated
      placement:
        constraints:
          - node.role == worker
      resources:
        limits:
          memory: 64M
        reservations:
          memory: 32M
      update_config:
        parallelism: 1
        order: start-first
        delay: 30s
    networks:
      - default

  cloudflared:
    deploy:
      mode: replicated
      replicas: 7
      placement:
        constraints:
          - node.role == worker
      resources:
        limits:
          memory: 64M
        reservations:
          memory: 32M
      update_config:
        parallelism: 1
        order: start-first
        delay: 30s
    image: cloudflare/cloudflared:latest
    command: tunnel --no-autoupdate run
    environment:
      TUNNEL_TOKEN:
    networks:
      - default
    
volumes:
  data:
    driver: local
    driver_opts:
      type: nfs4
      o: addr=10.156.0.60,rw
      device: ":/mnt/nfs_share/docker/volumes/n8n_data/_data"
  redis_data:
    driver: local
    driver_opts:
      type: nfs4
      o: addr=10.156.0.60,rw
      device: ":/mnt/nfs_share/docker/volumes/n8n_redis_data/_data"

networks:
  default:
  ```
  
  ```
DOMAIN_NAME=redacted
SUBDOMAIN=redacted
N8N_HOST=redacted
WEBHOOK_URL=redacted
WEBHOOK_TUNNEL_URL=redacted
SSL_EMAIL=redacted
DB_TYPE=postgresdb
DB_POSTGRESDB_DATABASE=n8n
DB_POSTGRESDB_HOST=redacted
DB_POSTGRESDB_USER=postgres
DB_POSTGRESDB_PASSWORD=redacted
POSTGRES_USER=${DB_POSTGRESDB_USER}
POSTGRES_PASSWORD=${DB_POSTGRESDB_PASSWORD}
POSTGRES_DB=${DB_POSTGRESDB_DATABASE}
POSTGRES_TAG=15-alpine
EXECUTIONS_MODE=queue
QUEUE_BULL_REDIS_HOST=n8n_redis
QUEUE_HEALTH_CHECK_ACTIVE=true
REDIS_PASSWORD=redacted
REDIS_TAG=7-alpine
N8N_BASIC_AUTH_ACTIVE=false
N8N_BASIC_AUTH_USER=user
N8N_BASIC_AUTH_PASSWORD=password
N8N_BLOCK_ENV_ACCESS_IN_NODE=true
N8N_DEFAULT_BINARY_DATA_MODE=default
N8N_PAYLOAD_SIZE_MAX=1024
N8N_PORT=5678
N8N_PROTOCOL=http
N8N_TAG=next
N8N_DEFAULT_LOCALE=pl
N8N_BASIC_AUTH_ACTIVE=false
NODE_ENV=production
NODE_FUNCTION_ALLOW_BUILTIN=nanoid
NODE_FUNCTION_ALLOW_EXTERNAL=nanoid
GENERIC_TIMEZONE=Europe/Warsaw
EXECUTIONS_DATA_MAX_AGE=336
EXECUTIONS_DATA_PRUNE=true
EXECUTIONS_DATA_SAVE_MANUAL_EXECUTIONS=true
EXECUTIONS_DATA_SAVE_ON_ERROR=all
EXECUTIONS_DATA_SAVE_ON_PROGRESS=false
EXECUTIONS_DATA_SAVE_ON_SUCCESS=none
EXECUTIONS_DATA_PRUNE_MAX_COUNT=100
N8N_EMAIL_MODE=smtp
N8N_ENCRYPTION_KEY=redacted
TUNNEL_TOKEN=redacted
```

docker swarm mode

**Comment by Shalashtein** [comment]
Can confirm my team's facing the same issue, we've recently had to scale of the setup of n8n as it has quickly reached its limits, we opted to use queue mode with workers and web hook processors following the docs, but webhook processors can't start the workflow when hitting the form url

The following setup uses the latest n8n docker images, organised into different dockerfiles to fit our deployment pipeline, we've tried several version but the gist is the same for everyone, we have:

- The main instance running on n8n.localhost (webhook processing disabled)
- webhook processors behind a load balancer on n8n.localhost/endpoints
- workers behind a load balancer

all in queue mode, connected to both postgres and redis
[docker compose file for reproduction](https://gist.github.com/Shalashtein/8447f533425389d3558969d11682475a)

Webhooks work fine, but forms don't here's an example url generated

n8n.localhost/endpoints/webhook/a975d106-8ee7-40f8-8d65-1f11bcf38ffc/n8n-form

and the response

{"code":0,"message":"Workflow Webhook Error: Workflow could not be started!"}

Not to sound pushy but would appreciate a quick investigation into this because we're trying to upgrade our license and setup n8n as a core offering in our tech stack, but this is blocking our basic setup of it

**Comment by janober** [comment]
Fix got released with [`n8n@1.40.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.40.0)
